src/Microsoft.DotNet.Build.Tasks/VersionTools/UpdateDependenciesAndSubmitPullRequest.cs

```diff
@@ -55,37 +57,40 @@ protected override void TraceListenedExecute()
                 {
                     item.SetMetadata(s_currentRef, masterSha);
                 }
+                string autoUpgradeBranch = item.GetMetadata(s_autoUpgradeBranch);
+                if (!string.IsNullOrEmpty(autoUpgradeBranch))
+                {
+                    item.SetMetadata(s_currentBranch, autoUpgradeBranch);
```

Comment:
It looks like it is used in BaseDependenciesTask but I didn't see it because I searched for s_currentBranch but you didn't use that constant there (which I left another comment about now).